di_website/common/blocks.py

```diff
@@ -110,6 +110,23 @@ class Meta:
         icon = 'fa-link'
         abstract = True
 
+class DatasourcesLinkBlock(StructBlock):
+    caption = CharBlock(
+        required=False,
+    )
+    page = PageChooserBlock(
+        required=False,
+        help_text='For the link/button to show, either this or the url are required'
+    )
+    url = URLBlock(required=False, help_text='An alternative to an internal page')
+
+    class Meta:
+        icon = 'fa-link'
+
+class DatasourcesStreamBlock(StreamBlock):
+    link = DatasourcesLinkBlock()
+    required = False
```

Comment:
If you're going to be using a StreamField, may as well reuse `LinkBlock`. This is an almost exact copy of it... The StreamField can be inlined in the model.

Reply:
Now resusing `AbstractLinkBlock`